### PR TITLE
Document function lifting

### LINKED ISSUE #10: Create documentation for lift operator

### PR DESCRIPTION
**Pull request by honzatrtik**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -346,6 +347,60 @@ TrySafe::fromCallable(fn() => $getTheUltimateAnswerOrThrow(true))
...
+```php
+use Bonami\Collection\Option;
+
+// Get our configuration value wrapped in Option, because it can be missing 
+function getConfig(string $key): Option {
+    $db = [
+        'host' => "domain.tld",
+        'port' => 8080,
+    ];
+    return Option::fromNullable($db[$key] ?? null);
```
Comment (jost125) [inline README.md]: Rename the $db variable to $config pls.

File: README.md [hunk 2]
```diff
@@ -346,6 +347,60 @@ TrySafe::fromCallable(fn() => $getTheUltimateAnswerOrThrow(true))
...
+
+// Unfortunately our string & int are wrapped in Option context, what now?
+$hostOption = getConfig('host');
+$portOption = getConfig('port');
+
+// Lifting method will allow to pass values wrapped in Option!
+$createUriLifted = Option::lift($createUri);
+
+// Return value is wrapped in Option too
+print $createUriLifted($hostOption, $portOption); // Some(https://domain.tld:8080)
```
Comment (jost125) [inline README.md:379]: // Option::some('https://domain.tld:8080')

File: README.md [hunk 3]
```diff
@@ -346,6 +347,60 @@ TrySafe::fromCallable(fn() => $getTheUltimateAnswerOrThrow(true))
...
+
+// Accepts list of strings, list of ints & returns list of uris created from all possible combinations  
+$createUriLifted = ArrayList::lift($createUri);
+
+print $createUriLifted($hosts, $ports); // [https://foo.org:80, https://bar.io:80, https://foo.org:8080, https://bar.io:8080]
+
+```
+
+As you can see, lifted function generates all possible combinations of hosts & ports for us, which can be very handy.
+
```
Comment (jost125) [inline README.md:418]: Please note, that empty `ArrayList` behaves similary greedy here - if any of arguments passed into lifted function is empty `ArrayList` the result will be empty `ArrayList`. Think of it like when you multiply numbers with `0` you wont get annithing else then `0` no matter how many times you multiply it with any non zero number.

See the similarity between `Option::none()`, `ArrayList::fromEmpty()`, `LazyList::fromEmpty()`, `TrySafe::failure($ex)`. All expresses "non-existence" of value and are analogous to `0` in context of multiplication (Cartesian multiplication for these type classes actually).

File: README.md (hunk 2 above)
Comment (honzatrtik) [reply README.md:379]: Well `Some(https://domain.tld:8080)` is the printed string here (see `Option::_toString` method implementation)

Comment (jost125) [reply README.md:379]: Ok then :-)

**Pull request merged**